### PR TITLE
Change how we determine if login is required

### PR DESCRIPTION
**Pull request by simo5**
This commit should address #188

### CODE REVIEW

File: tests/tbasic [hunk 1]
```diff
@@ -147,13 +147,13 @@ title PARA "Test fetching public keys without PIN"
 ORIG_OPENSSL_CONF=${OPENSSL_CONF}
 sed "s/^pkcs11-module-token-pin.*$/##nopin/" ${OPENSSL_CONF} > ${OPENSSL_CONF}.nopin
 OPENSSL_CONF=${OPENSSL_CONF}.nopin
-echo "" | ossl 'pkey -in $PUBURI -pubin -passin stdin -pubout -out ${TMPPDIR}/rsa.pub.nopin.pem'
-echo "" | ossl 'pkey -in $ECPUBURI -pubin -passin stdin -pubout -out ${TMPPDIR}/ec.pub.nopin.pem'
+echo "" | ossl 'pkey -in $PUBURI -pubin -pubout -out ${TMPPDIR}/rsa.pub.nopin.pem'
+echo "" | ossl 'pkey -in $ECPUBURI -pubin -pubout -out ${TMPPDIR}/ec.pub.nopin.pem'
```
Comment (Jakuje) [inline tests/tbasic:151]: do we still need the `echo ""` here when no prompt should be shown?

**Review by Jakuje — APPROVED** [review]
aside of the comment in regards to the test code, looks good. Will test it tomorrow,

**Comment by mouse07410** [comment]
I think this PR can and should be merged. Thanks!

File: tests/tbasic (hunk 1 above)
Comment (simo5) [reply tests/tbasic:151]: Not needed indeed, remove and added separate PARA for prompting test

**Review by Jakuje — APPROVED** [review]

**Comment by Jakuje** [comment]
Tested with libssh and all looks good when I provide the expect hints. For CI results with this branch, see the following MR:

https://gitlab.com/libssh/libssh-mirror/-/merge_requests/334

**Pull request merged**